src/ert/_c_wrappers/job_queue/run_status.py

```diff
@@ -0,0 +1,36 @@
+from dataclasses import dataclass
+from typing import Optional
+
+
+@dataclass
+class RunStatus:
+    RUNNING = "RUNNING"
+    FINISHED = "FINISHED"
+    CANCELED = "CANCELED"
+    FAILED = "FAILED"
+
+    status: Optional[str] = None
+    stdoutdata: str = ""
+    stderrdata: str = ""
+
+    def start(self):
+        self.status = RunStatus.RUNNING
+
+    def cancel(self):
+        self.status = RunStatus.CANCELED
+
+    def finish(self, msg=""):
+        if isinstance(msg, bytes):
+            msg = msg.decode()
+        self.stdoutdata = msg
+        if self.status == RunStatus.RUNNING:
+            self.status = RunStatus.FINISHED
+
+    def error(self, msg=""):
+        if isinstance(msg, bytes):
+            msg = msg.decode()
+        self.status = RunStatus.FAILED
+        self.stderrdata = msg
+
+    def __eq__(self, other):
```

Comment:
This __eq__ has a bug, and also, dataclasses have a separate mechanism for tweaking __eq__ through the `field` function.

Reply:
It's not a bug it is a workaround to allow for code like `if export_job.run_status == RunStatus.FAILED:` in place of `if export_job.run_status.status == RunStatus.FAILED:`

Reply:
That is very confusing though :D

Reply:
Generally, a reader will assume, that `a == b` will have the same value as `b==a` and this doesn't do that.

Reply:
These are some excellent points, and I don't like overwriting internal class functionality, but sometimes I forget the mantra that if you can do something it does not mean you should do it :)).  I will add the extra lines to do the proper status checks.